pkg/controller/test_helper.go

```diff
@@ -65,6 +67,38 @@ func readFile(filename string) (string, error) {
 	return str, nil
 }
 
+// WaitLoadtestFunc handles conditional watch events
+func WaitLoadtestFunc(event watch.Event) (bool, error) {
+	switch event.Type {
+	case watch.Added:
+		return true, nil
+	case watch.Modified:
+		return true, nil
+	default:
+		return false, nil
+	}
+}
+
+// WaitLoadtest waits until Loadtest resources exists
+func WaitLoadtest(clientSet clientSetV.Clientset, loadtestName string) error {
```

Comment:
if this function is used only in tests (according to file name) then it would be better not to return anything, but pass `t *testing.T` as the first parameter, then call `t.Helper()` in the first line to let testing know this is not really test and fail right here instead of returning an error and checking in the caller if error should always lead to test failure.

Reply:
I think this file, the way it is now, can be easily converted to a generic helper (instead of a test_helper)

Reply:
Personally I think the current setup is ok. Mostly because `test_helper.go` is part of the controller package and not postfixed with `_test` causing it it be part of the build of kangal. 
This makes adding the `testing` dependency more of an issue then having something that can test if something was loaded.

It would probably be better to not even have this `test_helper` but using a `internal/test` package which contains something like the `resources.go` or `watch.go` file.
In order to avoid this PR going out of scope I think we can merge and fix the `test_helper` issue another day.